package com.social.media.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SocialProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @OneToOne
    @JoinColumn(name="social_user_id")
    @JsonIgnore
    private SocialUser socialUser;

    public void setSocialUser(SocialUser socialUser) {
        socialUser.setSocialProfile(this);
        if(socialUser.getSocialProfile() !=this){
            this.socialUser = socialUser;
        }
    }
}
